test/images/agnhost/porter/porter.go

```diff
@@ -94,7 +120,19 @@ func serveTLSPort(port, value string) {
 	s := &http.Server{
 		Addr: "0.0.0.0:" + port,
 		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			fmt.Fprint(w, value)
+			body := value
+			if JSONResponse {
+				w.Header().Set("Content-Type", "application/json")
```

Comment:
since we're sending plaintext on error condition, you probably want to wait on doing this until just before the `body = string(j)` line